src/generic/codec.rs

```diff
@@ -239,11 +241,11 @@ where T: Decoder,
                 0 => false,
                 1 => {
                     trace!("message compressed, compression not supported yet");
-                    return Err(Status::UNIMPLEMENTED);
+                    return Err(::Error::Protocol(ProtocolError::Unsupported("compression not supported yet")));
```

Comment:
Any reason these shouldn't be specific variants also?

Reply:
No reason in particular, I can pull them up a level!

Reply:
Sorry, I meant variants on `ProtocolError`, not on the top level `Error`.